samtranslator/swagger/swagger.py

```diff
@@ -620,15 +624,15 @@ def set_path_default_apikey_required(self, path):
         :param string path: Path name
         """
 
+        valid_non_http_method_sections = ["parameters", "summary", "description", "$ref", "servers"]
```

Comment:
This seems to match the OpenAPI 3.0. Looking at references, this class seems to be specific to Rest APIs. Rest APIs support both [OpenAPI 2.0](https://swagger.io/specification/v2/) and 3.0: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-export-api.html

I think we may need to think deeper about this and the recent revert: https://github.com/aws/serverless-application-model/pull/2021 as a swagger validation overhaul.

Reply:
This is a block-list while #2021 was an allow-list. I mean, we know as a **fact** that `["parameters", "summary", "description", "$ref", "servers"]` are not API operations (HTTP methods)

Reply:
Maybe asking this a different way. If a customer is using OpenAPI 2.0 and specifies "servers" in the path object, should SAM fail with this being invalid? I know this just ignores the values and moves on, but not sure where we should fail this.

If SAM doesn't fail in the case above, then API Gateway should reject it. Which could be enough, just poking at this to see if we really need or want to do these deeper level of validation or not.

Reply:
Are you asking whether we should validate that the API definition complies with the OpenApi specification or not?
If so, I think this is not the responsibility of SAM. Here, we are trying to do our best to raise 4xx error instead of 5xx when applicable. So, if the API definition is invalid but it doesn't cause SAM to fail, I believe it shouldn't be a concern for SAM. Agree?

Reply:
I assumed this was in part validation we do.

Your reasoning makes sense to me.